agent/php_execute.c

```diff
@@ -1221,20 +1181,33 @@ static inline void nr_php_execute_segment_end(
   if (create_metric || (duration >= NR_PHP_PROCESS_GLOBALS(expensive_min))
       || nr_vector_size(stacked->metrics) || stacked->id || stacked->attributes
       || stacked->error) {
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO \
+    && !defined OVERWRITE_ZEND_EXECUTE_DATA
+
+    nr_php_execute_segment_add_metric(stacked, metadata, create_metric);
+    if (NULL == stacked->attributes) {
+      stacked->attributes = nr_attributes_create(NRPRG(txn)->attribute_config);
+    }
+
+    nr_php_txn_add_code_level_metrics(stacked->attributes, metadata);
+    nr_segment_t* s = nr_php_stacked_segment_move_to_heap(stacked TSRMLS_CC);
+
+#else
```

Comment:
A brief comment describing why this is done this way for OAPI & PHP8+ vs other configurations would be valuable in the future

Reply:
Added comment: [dcd676d](https://github.com/newrelic/newrelic-php-agent/pull/580/commits/dcd676dcb43e52f7d69a6334e6afb4aa8b537271)

Reply:
Awesome!